src/utils/daos_hdlr.c

```diff
@@ -203,20 +473,343 @@ pool_query_hdlr(struct cmd_args_s *ap)
+		fprintf(stderr, "size required to get attributes has raised, value has been truncated\n");
+	D_PRINT("%s\n", buf);
+
+out:
+	return rc;
+
+}
+
+int
+cont_list_attrs_hdlr(struct cmd_args_s *ap)
+{
+	size_t				 size, total_size, expected_size,
+					 cur = 0, len;
+	char				*buf;
+	int				rc = 0;
+
+	/* evaluate required size to get all attrs */
+	total_size = 0;
+	rc = daos_cont_list_attr(ap->cont, NULL, &total_size, NULL);
+	if (rc != 0) {
+		fprintf(stderr, "cont list attr failed: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	D_PRINT("Container attributes:\n");
+	if (total_size == 0) {
+		D_PRINT("No attributes\n");
+		D_GOTO(out, rc);
+	}
+
+	D_ALLOC(buf, total_size);
+	if (buf == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	expected_size = total_size;
+	rc = daos_cont_list_attr(ap->cont, buf, &total_size, NULL);
+	if (rc != 0) {
+		fprintf(stderr, "cont list attr failed: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	if (expected_size < total_size)
+		fprintf(stderr, "size required to gather all attributes has raised, list has been truncated\n");
+	size = min(expected_size, total_size);
+	while (cur < size) {
+		len = strnlen(buf + cur, size - cur);
+		if (len == size - cur) {
+			fprintf(stderr,
+				"end of buf reached but no end of string encountered, ignoring\n");
+			break;
+		}
+		D_PRINT("%s\n", buf + cur);
+		cur += len + 1;
+	}
+
+out:
+	return rc;
+
+}
+
+/* cont_get_prop_hdlr() - get container properties */
+int
+cont_get_prop_hdlr(struct cmd_args_s *ap)
+{
+	daos_prop_t		*prop_query;
+	struct daos_prop_entry	*entry;
+	int			rc = 0;
+
+	prop_query = daos_prop_alloc(0);
+	if (prop_query == NULL)
+		return -DER_NOMEM;
+
+	rc = daos_cont_query(ap->cont, NULL, prop_query, NULL);
+	if (rc) {
+		fprintf(stderr, "Container query failed, result: %d\n", rc);
+		D_GOTO(err_out, rc);
+	}
+
+	D_PRINT("Pool properties :\n");
+
+	entry = daos_prop_entry_get(prop_query, DAOS_PROP_CO_LABEL);
+	if (entry == NULL || entry->dpe_str == NULL) {
+		fprintf(stderr, "label property not found\n");
+		D_GOTO(err_out, rc = -DER_INVAL);
+	}
+	D_PRINT("label -> %s\n", entry->dpe_str);
+
+	entry = daos_prop_entry_get(prop_query, DAOS_PROP_CO_LAYOUT_TYPE);
+	if (entry == NULL) {
+		fprintf(stderr, "layout type property not found\n");
+		D_GOTO(err_out, rc = -DER_INVAL);
+	}
+	D_PRINT("layout type -> "DF_U64"\n", entry->dpe_val);
+
+	entry = daos_prop_entry_get(prop_query, DAOS_PROP_CO_LAYOUT_VER);
+	if (entry == NULL) {
+		fprintf(stderr, "layout version property not found\n");
+		D_GOTO(err_out, rc = -DER_INVAL);
+	}
+	D_PRINT("layout version -> "DF_U64"\n", entry->dpe_val);
+
+	entry = daos_prop_entry_get(prop_query, DAOS_PROP_CO_CSUM);
+	if (entry == NULL) {
+		fprintf(stderr, "checksum type property not found\n");
+		D_GOTO(err_out, rc = -DER_INVAL);
+	}
+	D_PRINT("checksum type -> "DF_U64"\n", entry->dpe_val);
+
+	entry = daos_prop_entry_get(prop_query, DAOS_PROP_CO_CSUM_CHUNK_SIZE);
+	if (entry == NULL) {
+		fprintf(stderr, "checksum chunck-size property not found\n");
```

Comment:
should it be chunk-size instead of chunck-size? Here and on line 754 below

Reply:
pure french speaker typo ;-) will be in next commit